apis/python/src/tiledbsoma/tiledb_object.py

```diff
@@ -49,6 +62,7 @@ def __init__(
             self._context = context or SOMATileDBContext()
 
         self._metadata = MetadataMapping(self)
+        self._close_stack = ExitStack()
```

Comment:
AFAICT this is not really needed; at most a single object is ever added to the stack.

Reply:
@gsakkis That is true considering this diff in isolation but,

1. Arrays: the C++ SOMAReader currently has a totally separate TileDB handle from the python code; this PR reuses the latter and only anticipates reusing the former, because [some C++ refactoring](https://github.com/single-cell-data/TileDB-SOMA/pull/730#issuecomment-1397493151) is needed (in follow-up PR).
2. Collections: when we write to a collection we then also refresh our cached view of it. But, the tiledb.Group handle opened in write mode doesn't allow us to read it, so we need to open a separate handle in read mode to do the cache refresh. Currently we are not reusing that read-mode handle, but it'd be nice to as another minor follow-up improvement.

Reply:
There is also that ExitStack helps to ensure reasonable things happen when exceptions occur in/around the unwind.